src/MetaCompilation/MetaCompilation/MetaCompilation/DiagnosticAnalyzer.cs

```diff
@@ -1302,317 +1294,320 @@ private void IfDiagnostic(CompilationAnalysisContext context, StatementSyntax st
             private bool CheckDiagnosticCreation(CompilationAnalysisContext context, SyntaxToken statementIdentifierToken, SyntaxToken keywordIdentifierToken, List<string> ruleNames, SyntaxList<StatementSyntax> statements, ParameterSyntax contextParameter)
             {
                 int statementCount = statements.Count;
-
-                SyntaxToken openParenToken = OpenParenAnalysis(statementIdentifierToken, statements);
-                if (openParenToken.Text == "")
+                if (statementCount < 9)
                 {
-                    IfDiagnostic(context, statements[3], OpenParenIncorrectRule, statementIdentifierToken.Text);
-                    return false;
-                }
-
-                if (statementCount > 4)
-                {
-                    SyntaxToken startToken = StartAnalysis(keywordIdentifierToken, statements);
-                    if (startToken.Text == "")
+                    List<string> diagnosticInfo = DiagnosticCreationCheck(ruleNames, statements[statementCount - 1]);
+                    if (diagnosticInfo.Count == 0 || diagnosticInfo[0] == "" || diagnosticInfo[1] == "")
                     {
-                        IfDiagnostic(context, statements[4], StartSpanIncorrectRule, keywordIdentifierToken.Text);
+                        IfDiagnostic(context, statements[statementCount - 1], DiagnosticIncorrectRule, ruleNames[0]);
```

Comment:
@daking2014 I use IfDiagnostic when there are incorrect statement because it accounts for the possibility that the incorrect statement could be an if-statement. The only major difference between IfDiagnostic and ReportDiagnostic is that IfDiagnostic will correctly squiggle if-statements (and therefore is only necessary when there is an incorrect statement as opposed to a missing statement)